staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -131,20 +131,42 @@ type Config struct {
 	Timeout time.Duration
 
 	// Dial specifies the dial function for creating unencrypted TCP connections.
+	// Deprecated: Use Dialer instead, which allows the transport to be cached.
+	// If both are set, Dialer takes priority.
```

Comment:
I don't think we should allow both being set if we can't introspect `Dialer` and make sure it matches `Dial` ... setting both is a recipe for confusion/error, in my opinion

Reply:
same comment applies to the other dual interface fields

Reply:
/hold

that is game over here :smile:

Reply:
/close